src/main/java/edu/gwu/csci6212/WildCardChar.java

```diff
@@ -0,0 +1,97 @@
+package edu.gwu.csci6212;
+
+import java.util.*;
+import java.io.*;
+
+public class WildCardChar
+{
+    public static boolean isMatching(char[] c_str, char[] c_pattern, int p, int q)
+	{
+	    // If both input string and pattern reaches their end,
+		// return true
+		if ( q < 0 && p < 0) {
+			return true;
+		}
+	    
+	    // empty pattern can only match with empty string 
+        else if (q == 0) 
+            return (p == 0); 
+        
+        // If only pattern reaches its end, return false
+		else if (q < 0) 
```

Comment:
I think your indentation is messed up here. I would expect this to be the same scope as line 17.